src/ITILSolution.php

```diff
@@ -195,6 +195,32 @@ public function prepareInputForAdd($input)
             $this->item->getFromDB($input['items_id']);
         }
 
+        // Check ticket mandatory fields
+        $ticket = $this->item;
+        $tt = $ticket->getITILTemplateToUse(0, $ticket->fields['type'] ?? '', $ticket->fields['itilcategories_id'] ?? '', $ticket->fields['entities_id'] ?? '');
+        if (count($tt->mandatory)) {
+            $mandatory_missing = [];
+            $fieldsname        = $tt->getAllowedFieldsNames(true);
+            foreach ($tt->mandatory as $key => $val) {
+                if (
+                    (isset($ticket->fields[$key])
```

Comment:
If the field is `null`, then `isset($ticket->fields[$key]` will return false and this mandatory field will not be considered as missing.